internal/targz/targz.go

```diff
@@ -141,7 +141,7 @@ func writeNewFile(fpath string, in io.Reader, fi os.FileInfo, buffer []byte) err
 		return fmt.Errorf("%s: making directory for file: %v", fpath, err)
 	}
 
-	out, err := os.Create(fpath)
+	out, err := os.OpenFile(fpath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, fi.Mode())
```

Comment:
This seems to be already done below?

https://github.com/cirruslabs/cirrus-ci-agent/blob/8996f9a23116c73960944eb6e6ae5bb33137592a/internal/targz/targz.go#L150

Reply:
Oh. Then I don't know why it's not working for @AlexWayfer. @AlexWayfer  could you please create an issue in https://github.com/cirruslabs/cirrus-ci-docs with a minimal reproducible case?

Reply:
> Oh. Then I don't know why it's not working for @AlexWayfer. @AlexWayfer could you please create an issue in https://github.com/cirruslabs/cirrus-ci-docs with a minimal reproducible case?

Of course. Done: https://github.com/cirruslabs/cirrus-ci-docs/issues/1087